common/define-grammar.js

```diff
@@ -1605,3 +1605,21 @@ function pipeSep1(rule) {
 function ampSep1(rule) {
   return seq(rule, repeat(seq(token('&'), rule)));
 }
+
+/**
+ * Creates a regex that matches PHP identifiers.
+ *
+ * Note that PHP officially only supports the following character regex
+ * for identifiers: ^[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*$
+ * However, there is a "bug" in how PHP parses multi-byte characters that allows
+ * for a much larger range of characters to be used in identifiers.
+
+ * @see https://www.php.net/manual/en/language.variables.basics.php
+ *
+ * @return {RexExp}
+ */
+function name() {
+  // We need to side step around the whitespace characters in the extras array.
+  const range = String.raw`\u0080-\u009f\u00a1-\u200a\u200c-\u205f\u2061-\ufefe\uff00-\uffff`;
+  return new RegExp(`[_a-zA-Z${range}][_a-zA-Z${range}\\d]*`);
+}
```

Comment:
since it's just called once, maybe just move it to the rule itself?

```javascript
name: _ => {
  // We need to side step around the whitespace characters in the extras array.
  const range = String.raw`\u0080-\u009f\u00a1-\u200a\u200c-\u205f\u2061-\ufefe\uff00-\uffff`;
  return new RegExp(`[_a-zA-Z${range}][_a-zA-Z${range}\\d]*`);
  }
```

Of the sorts

Reply:
Ah good idea, I can do that.